docker/package/packages.py

```diff
@@ -169,8 +169,8 @@
 ]
 
 postinst_steps_common = """
-mkdir -p -m755 /var/lib/tezos || true
-useradd --home-dir /var/lib/tezos tezos || true
+[ ! -d "/var/lib/tezos/" ] && mkdir -p -m755 /var/lib/tezos || true
+! id -u tezos &> /dev/null && useradd --home-dir /var/lib/tezos tezos && usermod -aG plugdev tezos || true
```

Comment:
Because `ledger_udev_postinst` also used by `tezos-client`, which does not create `tezos` user and thus it can't add it to the group

Reply:
It's necessary because actually commands fails if condition fails, failing installation

Reply:
> which does not create tezos user and thus it can't add it to the group

But shouldn't the user be created before here anyway?

I mean, the difference AFAIU is that beforehand the command to create the `tezos` user was run either way (including for `tezos-client`) whereas now it's only run if the user doesn't already exists, but either way the user should exist once the `ledger_udev_postisnt` is performed, no?

BTW is this related to [the other change](https://github.com/serokell/tezos-packaging/pull/582/commits/7fb30f49fa50e8e6b77dc0a896a9809c7186203a#diff-3d36859fab7aad684c238ee548fb94bb23d13270ea3dd848a1f70189f84919a5R304) in this file made by this commit?

Reply:
> either way the user should exist once the ledger_udev_postisnt is performed, no?
Well, I don't think it's necessary. For example, user could playing around with michelson tutorial, using public node. And to use ledger they also doesn't have to know anything about `tezos` user, they would add their user to `plugdev` group, as it stated in documetatation for using ledger with tezos.

Reply:
> BTW is this related to [the other change](https://github.com/serokell/tezos-packaging/pull/582/commits/7fb30f49fa50e8e6b77dc0a896a9809c7186203a#diff-3d36859fab7aad684c238ee548fb94bb23d13270ea3dd848a1f70189f84919a5R304) in this file made by this commit?

Yes, to preserve right order

Reply:
I'm confused :confused: 

What I'm saying is this, currently (on `master`, without this PR) we have:
1. the `tezos` user being created as part of the installation of every one of our package
2. the `udev-rules` script which (among other things) adds the tezos user to the `plugdev` group. This is run as part of the post-installation script of some of out packages (those that can communicate with a ledger device).

Here, in this PR, AFAIU you are making a change to avoid adding the `tezos` user multiple time, so not to get multiple useless messages that the `tezos` user already exists.

Fine, sure, but my question is: why would that need to move the "add `tezos` user to the plugdev group" as part of the installation of all of our packages?

It could be kept as part of the `udev-rules` and nothing would change because:
1. the `tezos` user for sure will exist when this script is run, we can make that happen
2. the `tezos` user isn't added to the `plugdev` group if no package that can use the ledger gets installed

Is this description more clear? If I didn't miss anything I don't really get your answer ... :confused:

Reply:
> the tezos user being created as part of the installation of every one of our package

That is not true. In the order of `packages.py` file, the first package introducting `tezos` user is `tezos-node`.

Look what happens with fresh vm:
```
vagrant@ubuntu-bionic:~$ sudo apt install -y tezos-client
...
groupadd: group 'plugdev' already exists
usermod: user 'tezos' does not exist
Processing triggers for libc-bin (2.27-3ubuntu1.6) ...
vagrant@ubuntu-bionic:~$ 
```

Reply:
> Fine, sure, but my question is: why would that need to move the "add tezos user to the plugdev group" as part of the installation of all of our packages?

Now it's also part of installation of all of out packages, since ledger_udev_postinst is part of almost every package postinst step. But with changes of that PR it's done only once.

Reply:
> Now it's also part of installation of all of our packages, since ledger_udev_postinst is part of almost every package postinst step

OK, but I don't think that's the right way to go, because the binaries in most packages can't use the ledger at all and so don't need the `udev` post-installation for it.

For example, `tezos-node` cannot use a ledger device on its own, so I don't see why it should perform the `ledger_udev_postinst`.

> Look what happens with fresh vm

Indeed, in this case you've found a problem :slightly_smiling_face: 

However, IMO the solution is that the `tezos-client` package should add the `tezos` user too and not that all packages should add support for `ledger` devices with the `plugdev` group.

Reply:
Only `tezos-client` and `tezos-signer` could interact with ledger, right?

Reply:
Yes, from the documentation that seems to be correct :+1: